gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -464,3 +469,174 @@ def _get_variant_calls(self, record, formats):
+                      self._file_name, data_line, str(e))
+      return MalformedVcfRecord(self._file_name, data_line, str(e))
+
+  def _convert_to_variant(self, variant_proto):
+    # type: (nucleus.protos.variants_pb2.Variant) -> Variant
+    return Variant(
+        reference_name=variant_proto.reference_name,
+        start=variant_proto.start,
+        end=variant_proto.end,
+        reference_bases=(variant_proto.reference_bases
+                         if variant_proto.reference_bases != MISSING_FIELD_VALUE
+                         else None),
+        alternate_bases=list(variant_proto.alternate_bases),
+        names=variant_proto.names[0].split(';') if variant_proto.names else [],
+        quality=variant_proto.quality,
+        filters=map(str, variant_proto.filter),
+        info=self._get_variant_info(variant_proto),
+        calls=self._get_variant_calls(variant_proto))
+
+  def _get_variant_info(self, variant_proto):
+    info = {}
+    for k in variant_proto.info:
+      data = self._convert_list_value(variant_proto.info[k],
+                                      self._is_info_repeated(k))
+      # Avoid including missing flags as `false` or other fields valued as `[]`.
+      if not data:
+        continue
+      info[k] = data
+    return info
+
+  def _convert_list_value(self, list_values, is_repeated):
+    """Converts an object of ListValue to python native types.
+
+    if is_repeated is set the output will be a list otherwise a single value.
+    """
+    output_list = []
+    for value in list_values.values:
+      if value.HasField('null_value'):
+        output_list.append(value.null_value)
+      elif value.HasField('number_value'):
+        output_list.append(value.number_value)
+      elif value.HasField('int_value'):
+        output_list.append(value.int_value)
+      elif value.HasField('string_value'):
+        output_list.append(value.string_value)
+      elif value.HasField('bool_value'):
+        output_list.append(value.bool_value)
+      elif value.HasField('struct_value'):
+        output_list.append(value.struct_value)
+      elif value.HasField('list_value'):
+        output_list.append(self._convert_list_value(value.list_value, True))
+      else:
+        raise ValueError('ListValue object has an unexpected value: %s' % value)
+
+    if is_repeated:
+      return output_list
+    else:
+      if len(output_list) > 1:
+        raise ValueError('a not repeated field has more than 1 value')
+      if len(output_list) == 1:
+        return output_list[0]
+      else:
+        return None
+
+  def _get_variant_calls(self, variant_proto):
+    calls = []
+    for call_proto in variant_proto.calls:
+      call = VariantCall()
+      call.name = call_proto.call_set_name
+      if not call_proto.genotype:
+        call.genotype.append(MISSING_GENOTYPE_VALUE)
+      else:
+        call.genotype = list(call_proto.genotype)
+
+      if call_proto.is_phased:
```

Comment:
This is a slight change in logic: previously, we considered a variant phased if it contained a PS field regardless of whether it used `|`. Im in favor of keeping the old logic. What is your reasoning for changing it?